Add codespell support (config, workflow to detect/not fix) and make it fix some typos

yarikoptic opened the pull request:
More about codespell: https://github.com/codespell-project/codespell .

I personally introduced it to dozens if not hundreds of projects already and so far only positive feedback.

CI workflow has 'permissions' set only to 'read' so also should be safe.

yarikoptic committed:
Add github action to codespell master on push and PRs

yarikoptic committed:
Add rudimentary codespell config

yarikoptic committed:
Add pre-commit definition for codespell

yarikoptic commented:
Hi @Lestropie , should I update it or you would prefer to not have it here?

Lestropie commented:
No issue with it being included. But I need to understand what it's doing, and it also implicitly highlights that there are other CI checks that the repo would likely benefit from eg. hadolint, black, pylint.

yarikoptic committed:
[DATALAD RUNCMD] chore: run codespell throughout fixing a few typos interactively

=== Do not change lines below ===
{
 "chain": [],
 "cmd": "codespell -w -i 3 -C 4",
 "exit": 0,
 "extra_inputs": [],
 "inputs": [],
 "outputs": [],
 "pwd": "."
}
^^^ Do not change lines above ^^^

yarikoptic commented:
done! sorry for the delay
